src/api/dto/CalloutFormDto.ts

```diff
@@ -157,20 +164,24 @@ function ComponentType() {
       if (typeof component !== "object" || component === null)
         throw new Error("Component must be an object");
 
-      switch (true) {
-        case inputTypes.includes(component.type):
-          return plainToInstance(InputCalloutComponentDto, component);
-        case selectTypes.includes(component.type):
-          return plainToInstance(SelectCalloutComponentDto, component);
-        case radioTypes.includes(component.type):
-          return plainToInstance(RadioCalloutComponentDto, component);
-        case nestedTypes.includes(component.type):
-          return plainToInstance(NestableCalloutComponentDto, component);
-        case "content" === component.type:
-          return plainToInstance(ContentCalloutComponentDto, component);
-        default:
-          throw new Error("Unknown component type " + component.type);
+      switch (component.type) {
+        case CalloutComponentType.CONTENT:
+          return plainToInstance(CalloutComponentContentDto, component);
+        case CalloutComponentType.INPUT_SELECT:
+          return plainToInstance(CalloutComponentInputSelectDto, component);
+      }
+
+      if (calloutComponentInputSelectableTypes.includes(component.type)) {
```

Comment:
You could also use `isCalloutComponentOfBaseType(component, CalloutComponentBaseType.NESTABLE)` from the common package here. Same for the base types bellow

Reply:
Woooo :tada: